tests/libvmifact/factory.go

```diff
@@ -89,26 +86,13 @@ func NewAlpineWithTestTooling(opts ...libvmi.Option) *kvirtv1.VirtualMachineInst
 
 func NewGuestless(opts ...libvmi.Option) *kvirtv1.VirtualMachineInstance {
 	opts = append(
-		[]libvmi.Option{libvmi.WithResourceMemory(qemuMinimumMemory())},
+		[]libvmi.Option{archAware.withQEMUMinimumMemory()},
 		opts...)
 	return libvmi.New(opts...)
 }
 
-func qemuMinimumMemory() string {
-	if isARM64() {
-		// required to start qemu on ARM with UEFI firmware
-		// https://github.com/kubevirt/kubevirt/pull/11366#issuecomment-1970247448
-		const armMinimalBootableMemory = "128Mi"
-		return armMinimalBootableMemory
-	}
-	return "1Mi"
-}
-
-func cirrosMemory() string {
-	if isARM64() {
-		return "256Mi"
-	}
-	return "128Mi"
+func WithArchAwareMemory() libvmi.Option {
```

Comment:
I don't think we need this function, I'd call `archAware.withRequestedMemory()` directly like you called to `archAware.withQEMUMinimumMemory()`.

Reply:
I don't want to expose the `archAware` var. I prefer to supply an option. This is going to be used all over, and this is an envelop. The user should not know about the internal implementation, and then we would be able to change it without changing all the calls.